CRM/Core/Payment/Dummy.php

```diff
@@ -78,19 +65,19 @@ public function __construct($mode, &$paymentProcessor) {
    * @throws \Civi\Payment\Exception\PaymentProcessorException
    */
   public function doDirectPayment(&$params) {
+    $propertyBag = PropertyBag::cast($params);
     // Invoke hook_civicrm_paymentProcessor
     // In Dummy's case, there is no translation of parameters into
     // the back-end's canonical set of parameters.  But if a processor
     // does this, it needs to invoke this hook after it has done translation,
     // but before it actually starts talking to its proprietary back-end.
-    if (!empty($params['is_recur'])) {
-      $throwAnENoticeIfNotSetAsTheseAreRequired = $params['frequency_interval'] . $params['frequency_unit'];
+    if (!$propertyBag->getIsRecur()) {
+      $throwAnENoticeIfNotSetAsTheseAreRequired = $propertyBag->getRecurFrequencyInterval() . $propertyBag->getRecurFrequencyInterval();
     }
```

Comment:
This will throw an exception if not set

Reply:
Yeah - I think that's Ok in the dummy - it's kinda the point

Reply:
The second one needs to be `..unit()`

Reply:
opps

Reply:
fixed